File: src/main/java/ordilov/randomplay/track/domain/TrackServiceImpl.java
package ordilov.randomplay.track.domain;

import java.util.List;
import java.util.stream.Collectors;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class TrackServiceImpl implements TrackService {

  private final TrackStore trackStore;
  private final TrackReader trackReader;

  @Override
  public List<TrackInfo> addTracks(List<Track> tracks) {
    return trackStore.store(tracks).stream()
        .map(TrackInfo::new).collect(Collectors.toList());
  }
}

File: src/main/java/ordilov/lissn/track/domain/TrackServiceImpl.kt
package ordilov.lissn.track.domain;

import org.springframework.stereotype.Service

@Service
class TrackServiceImpl(
    val trackStore: TrackStore,
    val trackReader: TrackReader
) : TrackService {

    override fun addTracks(tracks: List<Track>): List<TrackInfo> {
        return trackStore.store(tracks)
            .map { TrackInfo(it.id!!, it.title, it.resourceId) }.toList()
    }
}